docs/design/better_coordination_multi_operator.md

```diff
@@ -0,0 +1,129 @@
+# Better coordination for multiple operator instances
+
+## Metadata
+
+* Authors: @johscheuer
+* Created: 2024-11-19
+* Updated: 2024-11-19
+
+## Background
+
+The current way to run multi-region FoundationDB deployments with the operator is to run multiple operator instances in different namespaces and/or in different Kubernetes clusters.
+This brings a few synchronization challenges as those different operator instances are not communication with each other and they are not sharing state.
+We already added the `LockClient` to synchronize the upgrade step for minor and major upgrades, where all processes must be restarted at once.
+The `LockClient` is also used to get a lease that allows the operator instance to perform certain actions, like exclusions, that should only be execute by one instance at a time.
+
+## General Design Goals
+
+This design will show a possible way to synchronize the `exclusion`, `inclusion` and `removal` of process groups across different operator instances for the multi-region setup.
+
+## Current Implementation
+
+Right now only the restart (bounce) of the processes during a minor or major version upgrade is coordinated.
+All other bounces, exclusions, inclusions or removals are not coordinated with the locking mechanism only one operator instance will perform the according action to it's "local" processes.
+In most cases this works fine but causes more disruptions than required as every operator instance is perform the same set of operations on the local set of processes.
+The missing synchronization between the operator instance can result in multiple coordinator changes during the rollout of a setting or during an upgrade.
+
+## Proposed Design
+
+The idea is to extend the existing `AdminClient` with some similar functionality from the `LockClient` to coordinate the following actions:
+
+- coordinator selection
+- exclusions
+- inclusions
+- bounces
+
+The following methods will be added:
```

Comment:
Can we add more detail on how these lists will be modeled in the keyspace, and what kind of transactional properties (if any) we intend to use to control concurrent updates?